apps/remote_control/lib/lexical/remote_control/search/fuzzy.ex

```diff
@@ -336,31 +337,63 @@ defmodule Lexical.RemoteControl.Search.Fuzzy do
   end
 
   defp build_filter_fn do
-    mix_app_names = fn ->
-      applications =
-        if Mix.Project.umbrella?() do
-          Map.keys(Mix.Project.apps_paths())
-        else
-          List.wrap(Mix.Project.config()[:app])
-        end
-
-      MapSet.new(applications)
-    end
-
-    app_names =
+    deps_directories =
       if Mix.Project.get() do
-        mix_app_names.()
+        deps_roots()
       else
-        {:ok, names} =
+        {:ok, deps_roots} =
           RemoteControl.Mix.in_project(fn _ ->
-            mix_app_names.()
+            deps_roots()
           end)
 
-        names
+        deps_roots
       end
 
-    fn mapped(application: app, subtype: subtype) ->
-      subtype == :definition and MapSet.member?(app_names, app)
+    fn
+      mapped(subtype: :definition, grouping_key: path) ->
+        # if we don't have an app name, just make sure we're not
+        # in what looks like a deps directory
+        not Enum.any?(deps_directories, &String.starts_with?(path, &1))
+
+      _ ->
+        false
     end
   end
+
+  defp deps_roots do
+    deps_roots(RemoteControl.get_project())
+  end
+
+  defp deps_roots(%Project{mix_project?: true} = project) do
+    # Note: This function assumes that the deps directories for all
+    # found projects is `deps`. Projects may override this directory
+    # and lexical won't understand this. This was done because loading
+    # each sub-project is expensive and changes our global directory.
+
+    root_deps_path = Mix.Project.deps_path()
+    {root_deps_path, Project.mix_exs_path(project)}
+
+    [Project.root_path(project), "**", "mix.exs"]
+    |> Path.join()
+    |> Path.wildcard()
+    |> Enum.map(&Path.absname/1)
+    |> Enum.reduce([], fn mix_exs_path, deps_paths ->
+      deps_dir =
+        mix_exs_path
+        |> Path.dirname()
+        |> Path.join("deps")
+
+      if File.exists?(deps_dir) do
+        [deps_dir | deps_paths]
+      else
+        deps_paths
+      end
+    end)
+    |> Enum.reverse()
+    |> Enum.uniq()
```

Comment:
A slight simplification of this pipeline. Also, I think maybe the `uniq` can be excluded? I could be missing something obvious, but it seems like there shouldn't be duplicates.
```suggestion
    |> Enum.map(fn relative_mix_path ->
      relative_mix_path
      |> Path.absname()
      |> Path.dirname()
      |> Path.join("deps")
    end)
    |> Enum.filter(&File.exists?/1)
```